### PR TITLE
Clarify sparse .transpose() return type in docstrings

### PR DESCRIPTION
**Pull request by milesvant**
Documentation for `cupyx.scipy.sparse.csc_matrix` and `cupyx.scipy.sparse.csr_matrix` currently does not specify that `.transpose(...)` changes the sparse representation.

### CODE REVIEW

File: cupyx/scipy/sparse/_csc.py [hunk 1]
```diff
@@ -341,7 +341,7 @@ def transpose(self, axes=None, copy=False):
                 Otherwise, it shared data arrays as much as possible.
 
         Returns:
-            cupyx.scipy.sparse.spmatrix: Transpose matrix.
+            cupyx.scipy.sparse.csr_matrix: Transpose matrix in CSR format.
```
Comment (takagi) [inline cupyx/scipy/sparse/_csc.py:344]: Would you use the same description as scipy? (The part after colon)

File: cupyx/scipy/sparse/_csr.py [hunk 2]
```diff
@@ -521,7 +521,7 @@ def transpose(self, axes=None, copy=False):
                 Otherwise, it shared data arrays as much as possible.
 
         Returns:
-            cupyx.scipy.sparse.spmatrix: Transpose matrix.
+            cupyx.scipy.sparse.csc_matrix: Transpose matrix in CSC format.
```
Comment (takagi) [inline cupyx/scipy/sparse/_csr.py:524]: ditto

File: cupyx/scipy/sparse/_csc.py (hunk 1 above)
Comment (milesvant) [reply cupyx/scipy/sparse/_csc.py]: copied from https://github.com/scipy/scipy/blob/686422c4f0a71be1b4258309590fd3e9de102e18/scipy/sparse/_base.py#L768

File: cupyx/scipy/sparse/_csr.py (hunk 2 above)
Comment (milesvant) [reply cupyx/scipy/sparse/_csr.py]: same as above

**Review by takagi — APPROVED** [review]

**Comment by takagi** [comment]
Thanks, @milesvant!

**Pull request merged**